packages/block-editor/src/store/selectors.js

```diff
@@ -1402,8 +1402,11 @@ export const getBlockInsertionPoint = createSelector(
 		return { rootClientId, index };
 	},
 	( state ) => [
-		state.insertionPoint,
-		state.selection.selectionEnd,
+		state.insertionPoint?.rootClientId,
+		state.insertionPoint?.index,
+		state.insertionPoint?.__unstableWithInserter,
+		state.insertionPoint?.operation,
```

Comment:
I don't understand why we need to do this, these are all the keys of the state no?

Reply:
Because theoretically, `insertionPoint` is an object, and by listing those here, we'll prevent recalculation if `state.insertionPoint` is assigned a new object with the same subfields. 

I don't have proof that this is causing an issue right now, the main motivation for this PR was `selectionEnd` causing the issue I described. So if you prefer omitting those, I'm fine with that. Then can re-add them if we find a practical need for that. 

WDYT?

Reply:
Yes, let's do that. Restore the insertion point and keep the change for selectionEnd. Thanks Marin

Reply:
Done in https://github.com/WordPress/gutenberg/pull/47489/commits/ab5ea8a0ccbd1fb5ddd72aea596ccb1153b750cb

Thanks for the thorough and in-depth reviews, @youknowriad ❤️